build/utils/manifest.go

```diff
@@ -0,0 +1,74 @@
+package utils
+
+import (
+	// sha1 is not cryptographically secure
+	// but we just want to generate a reproducible fast hash
+	// to compare the contents of the files
+	"crypto/sha1" // nolint:gosec
+	"encoding/hex"
+	"errors"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func GenerateManifest() string {
+	manifest := ""
+	_ = filepath.WalkDir(".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if !d.IsDir() && strings.HasSuffix(path, ".go") {
+			hash, err := insecureHashFileContent(path)
+			if err != nil {
+				return nil
+			}
+			manifest = manifest + hash + ":" + path + "\n"
+		}
+		return nil
+	})
+
+	return manifest
+}
+
+// insecureHashFileContent returns the SHA1 hash of the file content.
+// It is not cryptographically secure. do not use for anything else
+func insecureHashFileContent(path string) (string, error) {
+	// Handle hashing big files.
+	// Source: https://stackoverflow.com/q/60328216/1722542
+
+	f, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+
+	defer func() {
+		_ = f.Close()
```

Comment:
Risk of swallowing error. Please check the error

Reply:
fs.close only returns an error if the file was already closed which is much unlike that will happen here.

Reply:
I logged the error.